### DISCUSSION TITLE
Override url generation with dashes for action macros

### DISCUSSION THREAD

**Discussion by hfjallemark** [luckyframework/lucky#505] [General] [answered]
Currently urls are being generated with underscores when using `action` and `nested_action` macros. It would be great if it could either default to use dashes instead as it seems more common or possibly provide a way to configure the url generation convention on a per app basis.

I don't know if this is still applicable but using dashes over underscores could provide a SEO improvement: https://www.mattcutts.com/blog/dashes-vs-underscores.

**Comment by paulcsmith** [comment]
Some more links I found:

* https://stackoverflow.com/questions/119312/urls-dash-vs-underscore
* https://support.google.com/webmasters/answer/76329?hl=en
* https://magenticians.com/dash-or-underscore-preferred-in-urls/

So maybe dash should be the default and underscore should be the option. I don't feel super strongly either way, but I think the option should at least exist for people that need/want it

**Comment by hfjallemark** [comment]
Yep, I think that makes sense.

**Comment by paulcsmith** [comment, marked as answer]
To handle this we are instead going to remove route-inference so you can use dashes or underscores. We'll also make it so you can set up checks to enforce dashes or underscores at compile time to maintain consistency! Thanks for this idea!

See https://github.com/luckyframework/lucky/discussions/789
